src/Spectre.Console.Extensions.Logging/SpectreConsoleLogger.cs

```diff
@@ -1,65 +1,68 @@
+            if (Config.IncludePrefix)
+            {
+                var prefix = GetLevelMarkup(logLevel);
+
+                sb.Append(prefix);
             }
-        }
-        public IDisposable BeginScope<TState>(TState state) {
-            return null;
-        }
 
-        public bool IsEnabled(LogLevel logLevel) {
-            return logLevel >= _config.LogLevel;
-        }
+            sb.Append(Name);
 
-        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter) {
-            if (!IsEnabled(logLevel)) {
-                return;
-            }
-            if (_config.EventId == 0 || _config.EventId == eventId.Id) {
-                var prefix = _config.IncludePrefix
-                    ? GetLevelMarkup(logLevel)
-                    : string.Empty;
-                var categoryStr = _config.IncludeEventId
-                    ? _name + $"[grey][[{eventId.Id}]][/]"
-                    : _name;
-                _console.MarkupLine(prefix + categoryStr);
-                _console.MarkupLine(string.Empty.PadRight(6) + formatter(state, exception));
+            if (Config.IncludeEventId)
+            {
+                sb.Append("[grey][[");
+                sb.Append(eventId.Id);
+                sb.Append("]][/]");
             }
+
+            sb.AppendLine();
+
+            sb.Append("      ");
+            sb.Append(formattedLine);
+
+            return sb.ToString();
         }
-        private string GetLevelMarkup(LogLevel level) {
+
+        private const string Unknown = "[italic dim grey]unkn[/]: ";
+        private const string Trace = "[italic dim grey]trce[/]: ";
+        private const string Debug = "[dim grey]dbug[/]: ";
+        private const string Information = "[dim deepskyblue2]info[/]: ";
+        private const string Warning = "[bold orange3]warn[/]: ";
+        private const string Error = "[bold red]fail[/]: ";
+        private const string Critical = "[bold underline red on white]crit[/]: ";
+
+        private static string GetLevelMarkup(LogLevel level)
+        {
             return level switch
             {
-                LogLevel.Trace => "[italic dim grey]trce[/]: ",
-                LogLevel.Debug => "[dim grey]dbug[/]: ",
-                LogLevel.Information => "[dim deepskyblue2]info[/]: ",
-                LogLevel.Warning => "[bold orange3]warn[/]: ",
-                LogLevel.Error => "[bold red]fail[/]: ",
-                LogLevel.Critical => "[bold underline red on white]crit[/]: ",
-                _ => throw new ArgumentOutOfRangeException(nameof(level))
+                LogLevel.Trace => Trace,
+                LogLevel.Debug => Debug,
+                LogLevel.Information => Information,
+                LogLevel.Warning => Warning,
+                LogLevel.Error => Error,
+                LogLevel.Critical => Critical,
+                _ => Unknown
             };
         }
```

Comment:
Just so I know, is this change intended to solve an actual problem you're facing?

I get that it's probably slightly more performant but it's also (in my view) dramatically less readable, so I'm not sure if you're actually seeing an issue here or if you're just trying to preemptively optimise?